lib/workers/global/ssh_socket.ts

```diff
@@ -0,0 +1,40 @@
+import type { HostRule } from '../../types';
+import * as cp from 'child_process';
+import * as net from 'net';
+import Agent from 'ssh-agent-js/client/index';
+import { join } from 'upath';
+
+export class SshSocket {
+  agent_socket: string;
+  agent_process: cp.ChildProcess;
+  socket: net.Socket;
+  agent: Agent;
+
+  async start(cacheDir: string) {
+    this.agent_socket = join(
+      cacheDir,
+      '.ssh-agent.' + Math.floor(Math.random() * 999999) + 1
```

Comment:
do we really need a random file? it should ever only run one instance on cache dir.

Reply:
I did this for the tests. But I guess as they are not running parallel we don't need it. Done